### PR TITLE
Android compatible routing

### PR DESCRIPTION
**Pull request by hummus-erectus**
Fixes issue leading to critical failure on Android upon login.

Previous error message:
`Error: Attempted to navigate before mounting the Root Layout component. Ensure the Root Layout component is rendering a Slot, or other navigator on the first render.`

Update uses `Stack.Screen` for static routes in RootLayout and `Slot` in MainLayout.

### CODE REVIEW

File: %ProgramData%/Microsoft/Windows/UUS/State/_active.uusver [hunk 1]
```diff
@@ -1 +0,0 @@
-1308.2407.15042.0
```
Comment (jjmountain) [inline %ProgramData%/Microsoft/Windows/UUS/State/_active.uusver]: Probably best to stick an entry in .gitignore to make sure these don't get in again e.g.

%ProgramData%/

**Comment by jjmountain** [comment]
Screenshot 2024-09-17 at 22 19 33

This is what I see on iphone btw. When I try to go to start free trial (sign up page) it goes to a blank page.

Screenshot 2024-09-17 at 22 22 45

Perhaps this is not what you see on Android though?

**Comment by jjmountain** [comment]
We shouldn't really be showing the Landing Page at all on mobile though as that's just meant for the web. Instead we should show a splash screen for a few seconds before showing sign up / sign in buttons.

File: %ProgramData%/Microsoft/Windows/UUS/State/_active.uusver (hunk 1 above)
Comment (hummus-erectus) [reply %ProgramData%/Microsoft/Windows/UUS/State/_active.uusver]: Sure, should I do that as a commit on this or another branch?

**Comment by hummus-erectus** [comment]
> Screenshot 2024-09-17 at 22 19 33
> This is what I see on iphone btw. When I try to go to start free trial (sign up page) it goes to a blank page.
> 
> Screenshot 2024-09-17 at 22 22 45
> Perhaps this is not what you see on Android though?

No, on Android I get taken to the sign up page as expected. I can't debug for iOS so not sure what the best course of action is? I suppose for now I could work on mobile stylng on my branch - which works on Android - and then push those changes but not the routing changes so you can see them on iOS? Or is that also not good practice?

**Comment by jjmountain** [comment]
I see. I generally don't think it's good for us to work on parts of the app and approve pull requests that the other person doesn't have any visibility into. I can look at setting up the Android simulator this weekend, and after I've set it up and check it works on both Android and iOS I can approve this, or add to it. 

I'm thinking we should probably aim to make sure both iphone and android versions of stuff works before merging changes for native stuff. What do you think?

**Comment by hummus-erectus** [comment]
Native stuff in this case refers to specific features of ios and Android? If so then sure, I agree we should work first on everything that's shared and make sure it works on both. Unless you mean something else?

As I think I wrote sonewhere else, routing now generally works on Android with things as they are. One of the "weird" things I experienced also occurs with the preview deploy on Chrome on mobile, which is that sometimes pressing the navigation icons at the bottom of the screen in mobile view results in showing the appropriate page for a split second before displaying the default wheels screen (the one represented by the bicycle). Refreshing the page fixes this. Did you experience this?

**Comment by jjmountain** [comment]
The weird thing you mentioned is the bug we have with auth. It would be great if we can figure out exactly how to reproduce it locally as the first step to fixing it.

**Comment by hummus-erectus** [comment]
OK I will treat that as a priority.

As for this pull request, it seems we don't need it any more as I am no longer getting the errors that prompted it. Can I close it?

**Comment by jjmountain** [comment]
The bug is fixed with https://github.com/Wheely-Good/wheels-expo/pull/12 so no need to look into it

**Comment by jjmountain** [comment]
Ok let's close for now as we are also not going to focus on Android for the time being.

**Pull request closed without merging**